server/controllers/postOrder.js

```diff
@@ -0,0 +1,27 @@
+const addOrder = require('../database/queries/orderQuery');
+const addMeal = require('../database/queries/orderMeal');
+const orderSchema = require('../validation/orderSchema');
+
+module.exports = (req, res, next) => {
+  let allMeals;
+  req.body.tableNumber = req.user.tableNumber;
+  orderSchema
+    .validateAsync(req.body)
+    .then((data) => {
+      const {
+        createdAt, totalPrice, meals,
+      } = data;
+      allMeals = meals;
+      return addOrder(createdAt, totalPrice, req.user.tableNumber);
+    })
+    .then((order) => {
+      const { rows } = order;
+      const orderId = rows[0].id;
+      return Promise.all(allMeals.map((meal) => addMeal(orderId, meal)));
+    }).then(() => {
+      res.send({ statusCode: 201 });
```

Comment:
you're sending 201 in the body while 200 is sent in the header.

It's kinda weird because in the tests you are testing for 200 while you suppose to be sending 201